File: Ejercicios Corredor/Trabajo Corredor/JsonEjemplo.cs
using System.Text.Json;

public class JsonEjemplo {
    public string Usuario = "{\"nombre\":\"Kevin\",\"edad\":25}";
    public string Producto = "{\"nombre\":\"Laptop\",\"precio\":3000}";
    public string Empresa = "{\"nombre\":\"CocaCola\"}";
    public string Pais = "{\"nombre\":\"Colombia\"}";
    public string Ciudad = "{\"nombre\":\"Bogotá\"}";

    public string CrearUsuario(string nombre, int edad) =>
        JsonSerializer.Serialize(new { nombre, edad });

    public string ObtenerNombre(string json) =>
        JsonDocument.Parse(json).RootElement.GetProperty("nombre").GetString();

    public int ObtenerEdad(string json) =>
        JsonDocument.Parse(json).RootElement.GetProperty("edad").GetInt32();

    public bool TieneClave(string json, string clave) =>
        JsonDocument.Parse(json).RootElement.TryGetProperty(clave, out _);

    public int ContarClaves(string json) =>
        JsonDocument.Parse(json).RootElement.EnumerateObject().Count();
}

File: Ejercicios Corredor/Trabajo Corredor/JsonEjemplo.go
package main
import (
	"encoding/json"
	"fmt"
)

type Usuario struct {
	Nombre string `json:"nombre"`
	Edad   int    `json:"edad"`
}

type JsonEjemplo struct{}

func (j JsonEjemplo) CrearUsuario(nombre string, edad int) Usuario {
	return Usuario{Nombre: nombre, Edad: edad}
}
func (j JsonEjemplo) ObtenerNombre(u Usuario) string { return u.Nombre }
func (j JsonEjemplo) ObtenerEdad(u Usuario) int { return u.Edad }
func (j JsonEjemplo) AJson(u Usuario) string {
	data, _ := json.Marshal(u)
	return string(data)
}
func (j JsonEjemplo) ContarCampos(u Usuario) int { return 2 }

func main() {
	u := JsonEjemplo{}.CrearUsuario("Kevin", 25)
	fmt.Println(JsonEjemplo{}.AJson(u))
}
